mlxtend/frequent_patterns/tests/test_association_rules.py

```diff
@@ -55,7 +55,7 @@
 
 # fmt: off
 def test_default():
-    res_df = association_rules(df_freq_items, df, len(df))
+    res_df = association_rules(df_freq_items, len(df), df)
```

Comment:
If NaN values are not present, the unit tests don't need the extra `df` here though, correct? I wonder if we could have the unit tests that don't have NaN values as they were before but then have an additional one where `df` is provided as you showed just to check that it doesn't cause errors. What do you think?

Reply:
We can definitely do that, it is a great idea! I removed `df` from the signature but still kept `num_itemsets=len(df)` as it is necessary even for the test case when NaNs do not exist. I also added a `test_nullability()` method to test the case when null values exist. Let me know if you need anything else.

Reply:
Awesome, thanks for adding this. I'd say don't worry about the formatting. It's one of the caveats when using black. It's just in the tests so it's ok.